### PR TITLE
Update Makefiles & co. to match buildsystem update

### PR DESCRIPTION
**Pull request by Ghabry**
Done:
- [x] 3ds
- [x] Android
- [x] emscripten (autotools, nothing to do)
- [x] ios (autotools, nothing to do)
- [x] linux (autotools, nothing to do)
- [x] osx (autotools, nothing to do)
- [x] Vita
- [x] Wii
- [x] Switch

### CODE REVIEW

**Comment by carstene1ns** [comment]
This is really bad to review online :/

**Comment by Ghabry** [comment]
according to my local test all Player binaries compile

**Comment by Ghabry** [comment]
okay this is also ready, added [X] Switch.

File: builds/3ds/Makefile [hunk 1]
```diff
@@ -56,11 +56,12 @@ CFLAGS	:=	-g0 -Wall -O2 -mword-relocations \
...
-CFLAGS	+=	$(INCLUDE) -DARM11 -D_3DS -DSUPPORT_AUDIO \
-			-DNO_DEBUG -DHAVE_MPG123 -DWANT_FASTWAV \
-			-DHAVE_LIBSPEEXDSP -DHAVE_LIBSNDFILE -DHAVE_TREMOR \
-			-DHAVE_WILDMIDI
-#-DCITRA3DS_COMPATIBLE
+CFLAGS	+=	$(INCLUDE) -DARM11 -D_3DS -DNO_DEBUG \
+			-DHAVE_FREETYPE -DHAVE_HARFBUZZ -DSUPPORT_AUDIO \
+			-DHAVE_MPG123 -DHAVE_LIBSNDFILE -DHAVE_TREMOR \
+			-DHAVE_XMP -DHAVE_WILDMIDI \
+			-DHAVE_LIBSPEEXDSP -DWANT_FMMIDI -DWANT_FASTWAV
```
Comment (carstene1ns) [inline builds/3ds/Makefile]: Are we sure we want to stress the 3DS cpu with fmmidi?

File: builds/psvita/Makefile [hunk 2]
```diff
@@ -29,15 +30,22 @@ CXX     = $(PREFIX)-g++
...
-		-DWANT_FMMIDI -DWANT_FASTWAV \
+CPPFLAGS = -DPSP2 -DHAVE_FREETYPE -DHAVE_HARFBUZZ -DSUPPORT_AUDIO \
+		-DHAVE_MPG123 -DHAVE_LIBSNDFILE -DHAVE_OGGVORBIS \
+		-DHAVE_XMP -DHAVE_WILDMIDI -DHAVE_OPUS \
+		-DHAVE_LIBSPEEXDSP -DWANT_FMMIDI -DWANT_FASTWAV \
 		-I$(VITASDK)/arm-vita-eabi/include/liblcf \
 		-I$(VITASDK)/arm-vita-eabi/include/pixman-1 \
+		-I$(VITASDK)/arm-vita-eabi/include/freetype2 \
+		-I$(VITASDK)/arm-vita-eabi/include/libxmp-lite \
+		-I$(VITASDK)/arm-vita-eabi/include/opus \
```
Comment (carstene1ns) [inline builds/psvita/Makefile]: Why should our libs be part of the sdk?

**Comment by Ghabry** [comment]
-label:Needs rebase

**Pull request merged**